### ISSUE TITLE
cannot write to exchange calendar

### ISSUE THREAD

**Issue by mattstudley** [Ericsson/exchangecalendar#337]
Hi there

after a recent upgrade of Thunderbird within Ubuntu, I am unable to write to my exchange calendar.

Ubuntu 14.04
Thunderbird 38.2.0
Lightning 4.0.2
Ericsson EWS 3.4.0-rc1

Problem :  If I unset the calendar read-only property in the properties dialogue box, this has no effect and does not persist; the read-only property is still set when I reopen the dialogue box.

I would very much like to solve this problem as soon as possible!

Many thanks for any help you can give me.

Matt

**Comment by mattstudley** [comment]
Actually, I have just found out that this is an intermittent fault; but I have so far no idea what triggers it.  I have just managed to save a new event in the exchange calendar!

**Comment by lifesoccult** [comment]
I am experiencing something similar to this. If I start from scratch, I can add my Exchange calendar and it will work - it's read/write and my notifications work. If it simply close/open TB, the calendar switches to read-only, notifications stop working and each meeting has "(busy)" appended to the title with no notification icon (bell); OR it shows duplicates for each meeting, one with a notification icon and no "(busy)" appended, and the other with no notification icon and "(busy)" appended. In both cases, notifications don't work and the calendar shows as read-only.

This actually started before 3.4.0-rc1 when we migrated from Exchange 2010 to Exchange 2013. But TB and Lightning also got updated during that same time period so I cannot say with certainty that one specific item is the culprit.

Jonathan

UBU 14.04
TB 38.2.0
Lightning 4.0.2
EWS 3.4.0-rc1
Exchange 2013

**Comment by bavincen** [comment]
Please check your rights with Admin...

**Comment by lifesoccult** [comment]
What the crackers does that mean? I'm the owner of my own calendar. Pretty standard stuff when it comes to Exchange. What "permissions" are you expecting this to be affected by?

**Comment by bavincen** [comment]
you may check that calendar appears read-only or not.. try clearing cache will taht help?

**Comment by lifesoccult** [comment]
Ready only is unchecked but the lock icon shows up anyway. Cleared cache. And when I add the calendar initially, everything works OK. It's only after I close and re-open Thunderbird does the calendar go into read only state and notifications cease. Problem persists after cache clearing.

**Comment by muthusuba** [comment]
When the connection to the server goes off, the calendar becomes readonly. 
[In the calendar view, check the left pane - where there are calendars listed (with checkbox) - you would see a readonly icon]

**Comment by lifesoccult** [comment]
What are you guys talking about? I've used this add on for a long time with no problems, but since switching to Exchange 2013 it continues to work incorrectly! There is a problem with how it communicates to Exchange 13! I'll get logs. You guys are being lazy and blameful.

**Comment by mattstudley** [comment]
I deleted the config, got rid of local state, purged and reinstalled the
whole suit.  All working beautifully now.

Pip Pip!

Matt

On Sat, Sep 19, 2015 at 1:28 PM, lifesoccult <notifications@github.com>
wrote:

> What are you guys talking about? I've used this add on for a long time
> with no problems, but since switching to Exchange 2013 it continues to work
> incorrectly! There is a problem with how it communicates to Exchange 13!
> I'll get logs. You guys are being lazy and blameful.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Ericsson/exchangecalendar/issues/337#issuecomment-141660313>
> .
>



-- 
Dr Matthew E. Studley
Bristol Robotics Laboratory

"The road to happiness lies in two simple principles;
find what interests you and that you can do well,
and put your whole soul into it - every bit of energy
and ambition and natural ability you have."
John D. Rockefeller

**Comment by bavincen** [comment]
@mattstudley What is that local state? . you may help @lifesoccult to debug.

**Comment by lifesoccult** [comment]
Can confirm success after doing the same as @mattstudley. 
Steps taken:
1. Removed all extensions from Thunderbird
2. `sudo apt-get purge thunderbird*`
3. `rm -rf ~/.thunderbird/`
4. `rm -rf ~/.cache/thunderbird/`
5. Reinstall & reconfigure mail, calendar

**Comment by mattstudley** [comment]
No idea, mate - I deleted it all!

On Mon, Sep 21, 2015 at 7:04 AM, bavincen <notifications@github.com> wrote:

> @mattstudley <https://github.com/mattstudley> What is that local state? .
> you may help @lifesoccult <https://github.com/lifesoccult> to debug.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Ericsson/exchangecalendar/issues/337#issuecomment-141883732>
> .
>



-- 
Dr Matthew E. Studley
Bristol Robotics Laboratory

"The road to happiness lies in two simple principles;
find what interests you and that you can do well,
and put your whole soul into it - every bit of energy
and ambition and natural ability you have."
John D. Rockefeller

**Comment by lifesoccult** [comment]
Looks like I spoke too soon. My problem still persists afterwards. Not sure what the cause it. Could be this ExQuilla add-on for EWS-based mail access.

**Comment by Be-ing** [comment]
> When the connection to the server goes off, the calendar becomes readonly.
[In the calendar view, check the left pane - where there are calendars listed (with checkbox) - you would see a readonly icon]

No, this is not the issue. Hovering my cursor over the calendar says that it is connected to the Exchange server.

**Comment by Octalus** [comment]
Hey,
still have this issue with newest version 3.5.0 of today - hoped it is going away.
I purged, deleted deinstalled everything in ubuntu - fresh installation of newest TB 38.4 with lightning 4.0.4.1 and with or without email configured previously - my exchange calendar stays read-only and (Busy) in the state.

Any ideas?

**Comment by bavincen** [comment]
@Octalus I will definitely help you out.. before that please check you have
necessary permission on respective calendars..  this problem is not for
everyone..

**Comment by Octalus** [comment]
it is my firm calendar - in outlook everything works and i can write to it. is outlook also using the webservices or does it use another secret way next to it? maybe it is a right restriction on the WS side for me?

**Comment by Octalus** [comment]
but no... also via the OWA web interface i can edit my calendar and i believe that this also uses the webservices - so no rights restriction on server side.

**Comment by Be-ing** [comment]
> but no... also via the OWA web interface i can edit my calendar and i believe that this also uses the webservices - so no rights restriction on server side.

Likewise. I can add an event to my calendar with OWA and it quickly shows up in Thunderbird, but I can't write an event from Thunderbird.

**Comment by bavincen** [comment]
Ok any of you post this settings screenshot.
![ews-mail-foldersettings](https://cloud.githubusercontent.com/assets/8401561/12046464/75513e4e-aedf-11e5-865e-42cd83d35be2.png)

**Comment by Octalus** [comment]
![bildschirmfoto vom 2015-12-30 08 51 44](https://cloud.githubusercontent.com/assets/12777042/12047856/cfd3297a-aed2-11e5-8250-745664480a13.png)
Not found - yours look very different here.

**Comment by Octalus** [comment]
here is my configuration part - calendar entries are showing correctly with it:
![bildschirmfoto vom 2015-12-30 08 56 53](https://cloud.githubusercontent.com/assets/12777042/12047909/9d3632c2-aed3-11e5-8dab-c09b5036cadc.png)

**Comment by bavincen** [comment]
@Octalus can you try after changing your language to english us

Thanks

**Comment by Octalus** [comment]
pretty the same:
![screenshot from 2015-12-30 09 51 20](https://cloud.githubusercontent.com/assets/12777042/12048383/fff64bac-aeda-11e5-9540-cd49db7b7257.png)

**Comment by Octalus** [comment]
Ah and I also tried a recreation of calendar in english US. Was the same.

**Comment by bavincen** [comment]
@Octalus do you think the problem could be because it's a share calendar.
Or can Ibmp provide you patch so that we can debug it .

**Comment by Octalus** [comment]
but it should also work with a "share" calendar? i don't know if this should generate a problem.
Yes i can help with debugging!

**Comment by Octalus** [comment]
My email: Marcus_Hammer@gmx.de

**Comment by bavincen** [comment]
I had been using team calendar. It worked fine. Let me try it in latest.

**Comment by Be-ing** [comment]
Folder properties also shows "not found" at the top with all of the fields blank for me. I'd be happy to test experimental XPIs, but I don't know how to build a patched XPI.

**Comment by Be-ing** [comment]
When the calendar does work (which seems to be random and only changes when restarting Thunderbird), Folder properties shows:
![folder-properties](https://cloud.githubusercontent.com/assets/9455094/12054075/485253aa-aee5-11e5-98eb-0cb5d968145c.png)

**Comment by bavincen** [comment]
@Be
https://github.com/Ericsson/exchangecalendar/wiki/How-to-Install-latest-addon

**Comment by Octalus** [comment]
So, how can i help to debug the problem?

**Comment by Be-ing** [comment]
Okay, that looks easy enough @bavincen . Let me know when there is a build available to test.

**Comment by Octalus** [comment]
hey,

finally i was able to make it work/writable!
It seems that my calendar moved to another adress and i had a kind of link to the adress i used in the first place. There i had the read-only problem. if i use the other one it seems to work fine!

so maybe not a bug here!

Thanks for all - Great job with the Addon.

**Comment by bavincen** [comment]
That's a good one
Have a nice day

**Comment by Be-ing** [comment]
@Octalus, can you provide more details? How did you initially set up your calendar in Lightning and how did you get the new address? I used the autodiscovery function to set up my calendars.

**Comment by Octalus** [comment]
Hey, 

My firm changed a while ago the email pattern from lastname@firm.de to firstname.lastname@firm.de and configured a forwarding to and from both possibilities. I never recognized or changed anything because the old pattern is still working. But it seems that the new pattern is the "main" account and the old pattern only the forwarding. 

So I switched to the new pattern and this is working (until now I never took care of the new pattern) 

So maybe you have a similar problem? 

Regards 
Marcus

**Comment by campenberger** [comment]
for what it is worth, the email name is also case sensitive. IT hat my email in exchange with the first two characters capitalized. I did the primary email address in the EWS account all lower case and got all the symptoms described here. Fixing the capitalization seem to resolve it.

**Comment by mwoodiupui** [comment]
I ran into this in the past few days, tried many things that didn't work.  Today I played with the "Exchange (EWS) Properties", tried autodiscovery (failed), put everything back the way I had it, and hit "Save and Reset".  It pulled down about 2500 jobs and, after working through them, my Exchange calendar seems to be working properly and stably.

I had been having trouble for some time with events that couldn't be dismissed -- they would reappear instantly, sometimes even replacing themselves continuously until I shut down and restarted Thunderbird.  I think that upgrading to the latest of this plugin helped that, but there may have been damage in my calendar.
